### ISSUE TITLE
Generic error in an external library', ' Error initializing a simple  filtergraph', 'Error opening output file 

### ISSUE THREAD

**Issue by Paradoxum** [stevezau/plex_generate_vid_previews#39]
I managed to generate thumbs using the exact same docker file for all my libraries 1 by 1 by adding the paths each time, now I have only TV Shows left, and I am getting this:

![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/d5c92ef5-9ce4-4466-8d72-3394d35cfb69)

![s5wCGGb 1](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/9a4e7a44-c603-4d03-ab29-054ce7e32603)

It looks like it is able to generate some.. but errors on a lot of them. Even episodes from the same series/release torrent etc.
I tried clearing Plex's Cache Codecs and Drivers and letting it get them again which sometimes fixes an error, but not working now.

Any ideas?

Here's a little more, seems to generate some eps but that error on others;
![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/e9b55445-9d8d-4645-9b98-0d565e4b0e92)

https://stackoverflow.com/questions/41761684/avcodec-encode-video2-return-542398533-generic-error-in-an-external-library-onl

https://stackoverflow.com/questions/77751538/ffmpega-general-error-in-an-external-library-occurred-when-using-ffmpeg6-1s-av

https://github.com/nyanmisaka/ffmpeg-rockchip/issues/5

Never seen this error before. exact same docker file as other libraries. it seems able to do some episodes of the same series but fails on others?

* The script has general issues with needing permissions, so after running it I have to rush to plex and media shares and set them both r/w for all and that seems to fix the script

* I just did it now a few times, and i've got almost a whole log screen of successful generations

* Is there something I can put in the docker file to make it run as privileged / root ?

![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/d2d0e670-8a19-4045-8954-838c1ad442ba)

* After setting r/w on those dirs a couple times (something changes them I don't know what) seems to run fine with 0 errors so far. 

Edit: https://github.com/blakeblackshear/frigate/issues/690
Says something about the error being not able to find your GPU.

https://www.reddit.com/r/Tdarr/comments/qis38p/device_setup_failed_for_decoder_on_input_stream/

https://www.reddit.com/r/ispyconnect/comments/umpmj0/agent_dvr_cant_get_hw_decoding_working/

Strange that it doesn't work at first then does etc.

**Comment by Paradoxum** [comment]
nvidia-smi in the console for the script

![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/68921970-ff7a-485d-8162-af4eed64e806)

**Comment by Paradoxum** [comment]
It seemed to be running fine, then plex crashed:
![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/b410efb9-19e9-4c74-bb86-02e6707fe28e)

plex logs: [Logs.zip](https://github.com/user-attachments/files/15997667/Logs.zip)

**Comment by Paradoxum** [comment]
It was running without issue for a couple of hours then suddenly;

![vyxZ03v 1](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/59d37297-ef41-402a-b558-f4dc547d4d55)

**Comment by Paradoxum** [comment]
I think I have the root cause of this. The script becomes unable to write anything in the Plex share. I have to go here, outside of the shares:
![image](https://github.com/stevezau/plex_generate_vid_previews/assets/6451395/f172e687-790f-4718-b541-e2905209bcce)

and then set R/W all on the Plex share which contains all the plex metadata. Doing it on anything inside that doesn't work, the script needs R/W to the Plex container.

So, I also don't know why it keeps reverting my changes it must be another docker or even plex doing it, but how do I make the script run as like pure root so I don't even need to change permissions?

Thanks

**Comment by stevezau** [comment]
@Paradoxum getting your file permissions right is beyond the scope of this script.. 

But.. you might want to look at the --user option when starting the container.. Suggest you start it as the same user who owns the plex files.

**Comment by stevezau** [comment]
@Paradoxum i am going to close this ticket as i think your issues are no longer related to this script but more file system permission?

**Comment by Paradoxum** [comment]
I am not sure what happened, but I took my server down for a few days to sort it out in a fractal 7xl - as soon as I turned it on, the script  seems to be running fine without errors.

Could it be my GPU was not seated correctly? because I noticed that the M2 block I have on the drive right below it cause it to not fully go in. since I soprted that I don't think there's an issue.

![image](https://github.com/user-attachments/assets/2937f5ec-a80c-4f02-a6fd-94dc66a7330d)
![image](https://github.com/user-attachments/assets/a445609a-3571-4e46-820e-796ecfa0ad01)

plex using 200gb metadata and it's only on
